task/universal/ssh/connection.go

```diff
@@ -0,0 +1,44 @@
+package ssh
+
+import (
+	"time"
+
+	"golang.org/x/crypto/ssh"
+)
+
+func RunCommand(command string, timeout time.Duration, hostAddress string, userName string, privateKey string) (string, error) {
+	parsedPrivateKey, err := ssh.ParsePrivateKey([]byte(privateKey))
+	if err != nil {
+		return "", err
+	}
+
+	configuration := &ssh.ClientConfig{
+		User: userName,
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(parsedPrivateKey),
+		},
+		// Doesn't matter for this use case, but isn't a good practice either.
+		// FIXME: IT **DOES** MATTER NOW! FIND A SOLUTION ASAP!
```

Comment:
We only care if it changes over time right :thinking:

Reply:
This comment is because of `HostKeyCallback: ssh.InsecureIgnoreHostKey()` below: we don't have any [really] secure way of provisioning a fixed host key into instances, unless we use another cloud service like Secrets Manager.

Reply:
Here we're relying on [TOFU](https://en.wikipedia.org/wiki/Trust_on_first_use) or, rather, “TOEU” (trust on every use), rendering the identity check ineffective.

Unless we know the host key/fingerprint in advance, we're vulnerable to MITM when connecting to the instance.

Still, this prototype doesn't use SSH at all, and keys are being generated just in case users want to connect manually to the instance. 😄

Reply:
Added it to https://github.com/iterative/terraform-provider-iterative/pull/237#issue-1048177975 so we don't forget.